src/db.c

```diff
@@ -1712,6 +1712,21 @@ void deleteExpiredKeyAndPropagate(serverDb *db, robj *keyobj) {
     server.stat_expiredkeys++;
 }
 
+/* Delete the specified expired key from overwriting and propagate the DEL or UNLINK. */
+void deleteExpiredKeyFromOverwriteAndPropagate(client *c, robj *keyobj, int do_delete) {
```

Comment:
The difference between this function and deleteExpiredKeyAndPropagate is not very clear to me. When is each of these used? Maybe the name can be better or we can rename both functions if we can find a more clear purpose of each one with other names.

"From Overwrite" means that an explicit command that deletes the key?

It's a bit confusing to have a `do_delete` that can be false in a function that is called delete-something. Can we avoid it? What I can see, this function is always called with `do_delete = 1`.

Reply:
ohh, the do_delete use was removed in here: https://github.com/valkey-io/valkey/pull/865/commits/e9e089bcf4134b4add98616a01c726d3de943782

i forgot to remove it.

The `deleteExpiredKeyFromOverwriteAndPropagate` name was suggested by Madelyn